### PR TITLE
Fix file names, golang conventions, and example test.

### PR DESCRIPTION
**Pull request by maerics**
- [x] Rename `glided-rose` to `gilded_rose`.
- [x] Create `GildedRose` type, per other languages.
- [x] Export `Item` type per golang conventions.
- [x] Extract [`UpdateQuality` method](https://github.com/emilybache/GildedRose-Refactoring-Kata/blob/master/GildedRoseRequirements.txt#L31).
- [x] Provide sample (failing) test per other languages.
- [x] Update readme with go 1.5 (latest) test coverage tool usage.

@emilybache @ldez review please, thx.

### CODE REVIEW

**Comment by ldez** [comment]
hello,

In golang convention, It does not recommend using the underscore, you can use underscore in filename only for CPU dependent file or test.
The convention is : lowercase, without nothing between words.

see:
- https://github.com/golang/go/blob/master/src/cmd/fix/printerconfig.go

File: go/gilded_rose.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
+package main
+
+import "fmt"
+
+type GildedRose struct {
```
Comment (ldez) [inline go/gilded_rose.go]: this struct is not necessary.
It's not the code that goes straight to the point.
This struct can be made in the kata but not in the starter.

File: go/gilded_rose.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	Items []Item
+}
+
+type Item struct {
+	Name            string
+	SellIn, Quality int
+}
+
+func main() {
+	rose := GildedRose{
```
Comment (ldez) [inline go/gilded_rose.go]: Make GildedRose not testable is a (original kata) choice: you're first step must be to make code testable.

If you do make code already testable, you lost a part of the kata.

File: go/readme.md [hunk 3]
```diff
@@ -1,21 +1,19 @@
...
-- Run tests and coverage :
+- Run tests and coverage:
 
 ```shell
-go test -coverprofile=coverage.out
-
-go tool cover -html=coverage.out
-```
\ No newline at end of file
+go test -cover
```
Comment (ldez) [inline go/readme.md]: Provide the 2 syntaxes, seems more interesting to me.

File: go/gilded_rose.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+			Item{"Sulfuras, Hand of Ragnaros", 0, 80},
+			Item{"Backstage passes to a TAFKAL80ETC concert", 15, 20},
+			Item{"Conjured Mana Cake", 3, 6},
+		},
+	}
+	rose.UpdateQuality()
+	fmt.Print(rose.String())
+}
+
+func (gr *GildedRose) UpdateQuality() {
```
Comment (ldez) [inline go/gilded_rose.go]: use inheritence here, seems not a good way.

File: go/gilded_rose.go [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+				if item.Quality < 50 {
+					item.Quality = item.Quality + 1
+				}
+			}
+		}
+	}
+
+}
+
+func (gr *GildedRose) String() string {
```
Comment (ldez) [inline go/gilded_rose.go]: Use OOP style is not the Go approach

File: go/gilded_rose.go (hunk 4 above)
Comment (ldez) [inline go/gilded_rose.go]: In Go we prefer split data and behaviours

**Comment by maerics** [comment]
Ah, I see.  It seems like (some of) these changes should be first steps in the exercise, not the starter itself.

**Comment by emilybache** [comment]
Thanks for your thoroughly commented changes that explain why you've done what you've done. Seems reasonable to me

**Pull request closed without merging**